python/lsst/ts/externalscripts/auxtel/latiss_acquire.py

```diff
@@ -0,0 +1,604 @@
+            Script configuration, as defined by `schema`.
+        """
+
+        # Instantiate BestEffortIsr -
+        self.best_effort_isr = self.get_best_effort_isr()
+
+        # instantiate the quick measurement class
+        qm_config = QuickFrameMeasurementTask.ConfigClass()
+        self.qm = QuickFrameMeasurementTask(config=qm_config)
+
+        self.do_reacquire = config.do_reacquire
+
+        if not self.do_reacquire:
+            self.object_name = config.object_name
+
+        self.object_ra = config.object_ra
+        self.object_dec = config.object_dec
+
+        # config for the single image acquisition
+        self.acq_grating = config.acq_grating
+        self.acq_filter = config.acq_filter
+        self.acq_exposure_time = config.acq_exposure_time
+
+        # Max number of iterations to perform when putting source in place
+        self.max_acq_iter = config.max_acq_iter
+
+        # Tolerance in arcsec for distance from main source centroid to
+        # relevant sweet spot
+        self.target_pointing_tolerance = config.target_pointing_tolerance
+        self.target_pointing_verification = config.target_pointing_verification
+
+        self.reason = config.reason
+
+        self.program = config.program
+
+        self.time_on_target = config.time_on_target
+
+        if self.time_on_target is None:
+            self.time_on_target = 0.0
+            self.azimuth_wrap_strategy = WrapStrategy.MAXTIMEONTARGET
+        else:
+            self.azimuth_wrap_strategy = WrapStrategy.OPTIMIZE
+
+        self.rot_type = config.rot_type
+        self.rot_value = config.rot_value
+
+        self.do_manual_final_position = config.do_manual_final_position
+
+        if self.do_manual_final_position:
+            self.target_position = self.do_manual_final_position
+        else:
+            self.target_position = latiss_constants.sweet_spots[self.acq_grating]
+
+        self.image_in_oods_timeout = 15.0
+
+    def set_metadata(self, metadata):
+        slew_duration = 0 if self.do_reacquire else 60
+        latiss_config_duration = 5
+        acq_iter_overhead_duration = 5
+
+        metadata.duration = (
+            slew_duration
+            + latiss_config_duration
+            + (self.acq_exposure_time + acq_iter_overhead_duration) * self.max_acq_iter
+        )
+
+    # To-Do: move to new class as part of DM-37665
+    def get_best_effort_isr(self):
+        # Isolate the BestEffortIsr class so it can be mocked
+        # in unit tests
+        return BestEffortIsr()
+
+    # TODO: move to new class as part of DM-37665
+    async def get_next_image_data_id(self, flush=True):
+        """Return dataID of image that appears from the ATOODS CSC.
+
+        This is meant to be called at the same time as a take image command.
+        If this is called after take_image is completed, it may not receive
+        the imageInOODS event.
+
+        Parameters
+        ----------
+        timeout: `float`
```

Comment:
This is not a parameter of this method. But `flush` is and it is not mentioned here.

Reply:
Opted to remove flush parameter from method and set to True in call.